component/src/main/java/org/wso2/extension/siddhi/io/feed/sink/FeedSink.java

```diff
@@ -0,0 +1,268 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.extension.siddhi.io.feed.sink;
+
+import org.apache.abdera.Abdera;
+import org.apache.abdera.model.Document;
+import org.apache.abdera.model.Entry;
+import org.apache.abdera.protocol.client.AbderaClient;
+import org.apache.abdera.protocol.client.ClientResponse;
+import org.apache.commons.httpclient.UsernamePasswordCredentials;
+import org.wso2.extension.siddhi.io.feed.sink.exceptions.FeedErrorResponseException;
+import org.wso2.extension.siddhi.io.feed.utils.Constants;
+import org.wso2.extension.siddhi.io.feed.utils.EntryUtils;
+import org.wso2.siddhi.annotation.Example;
+import org.wso2.siddhi.annotation.Extension;
+import org.wso2.siddhi.annotation.Parameter;
+import org.wso2.siddhi.annotation.util.DataType;
+import org.wso2.siddhi.core.config.SiddhiAppContext;
+import org.wso2.siddhi.core.exception.ConnectionUnavailableException;
+import org.wso2.siddhi.core.stream.output.sink.Sink;
+import org.wso2.siddhi.core.util.config.ConfigReader;
+import org.wso2.siddhi.core.util.transport.DynamicOptions;
+import org.wso2.siddhi.core.util.transport.OptionHolder;
+import org.wso2.siddhi.query.api.definition.StreamDefinition;
+import org.wso2.siddhi.query.api.exception.SiddhiAppValidationException;
+
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * The sink implementation of feed.
+ */
+@Extension(
+        name = "feed",
+        namespace = "sink",
+        description = " The feed sink allows to publish atom feed entries to atom implemented http servers. " +
+                "The stream variables should be the standard element name of the entry of atom document. 'id'," +
+                " 'title', 'link', 'updated', 'author', 'published', 'summery' are the supported elements.",
+        parameters = {
+                @Parameter(name = Constants.URL,
+                        description = "The feed end point url",
+                        type = DataType.STRING),
+                @Parameter(name = Constants.ATOM_FUNC,
+                        description = "Atom function of the request. " +
+                                "Acceptance parameters are 'create', 'delete', 'update'",
+                        type = DataType.STRING),
+                @Parameter(name = Constants.USERNAME,
+                        description = "User name of the basic authentication",
+                        optional = true,
+                        defaultValue = Constants.CREDENTIALS,
```

Comment:
Update the names of the variables used as defaults from the Constants class.
eg: Since you are using Constants.CREDENTIALS as the default value for the Username, update Constants.CREDENTIALS as DEFAULT_USERNAME

Reply:
Revisit all constant namings